pkg/scheduler/extender/extender.go

```diff
@@ -257,105 +261,212 @@ func (e *Extender) gatherVolumesByProvisioner(ctx context.Context, pod *coreV1.P
-	failedNodesMap schedulerapi.FailedNodesMap, err error) {
-	if len(volumes) == 0 {
-		return nodes, failedNodesMap, err
+// filteredNodes - represents the filtered out nodes, with node names and failure messages
+func (e *Extender) filter(ctx context.Context, pod *coreV1.Pod, nodes []coreV1.Node, capacities []*genV1.CapacityRequest) (matchedNodes []coreV1.Node,
+	filteredNodes schedulerapi.FailedNodesMap, err error) {
+	// ignore when no storage allocation requests
+	if len(capacities) == 0 {
+		return nodes, nil, nil
 	}
 
-	// TODO: do not read all ACs and ACRs for each request: https://github.com/dell/csi-baremetal/issues/89
-	acReader := capacityplanner.NewACReader(e.k8sClient, e.logger, true)
-	acrReader := capacityplanner.NewACRReader(e.k8sClient, e.logger, true)
-	reservedCapReader := capacityplanner.NewUnreservedACReader(e.logger, acReader, acrReader)
-	capManager := e.capacityManagerBuilder.GetCapacityManager(e.logger, reservedCapReader)
-
-	placingPlan, err := capManager.PlanVolumesPlacing(ctx, volumes)
+	// construct ACR name
+	reservationName := getReservationName(pod)
+	// read reservation
+	reservation := &acrcrd.AvailableCapacityReservation{}
+	err = e.k8sClient.ReadCR(ctx, reservationName, "", reservation)
 	if err != nil {
-		return matchedNodes, failedNodesMap, err
+		if k8serrors.IsNotFound(err) {
+			// create new reservation
+			if err := e.createReservation(ctx, pod.Namespace, reservationName, nodes, capacities); err != nil {
+				// cannot create reservation
+				return nil, nil, err
+			}
+			// not an error - reservation requested
+			return nil, nil, nil
+		}
+		// issue with reading reservation
+		return nil, nil, err
 	}
 
-	noACForNodeMsg := "Node doesn't contain required amount of AvailableCapacity"
+	// reservation found
+	return e.handleReservation(ctx, reservation, nodes)
+}
 
-	failedNodesMap = schedulerapi.FailedNodesMap{}
-	for _, node := range nodes {
-		if placingPlan == nil {
-			failedNodesMap[node.Name] = noACForNodeMsg
-			continue
-		}
+func getReservationName(pod *coreV1.Pod) string {
+	namespace := pod.Namespace
+	if namespace == "" {
+		namespace = "default"
+	}
+
+	return namespace + "-" + pod.Name
+}
+
+func (e *Extender) createReservation(ctx context.Context, namespace string, name string, nodes []coreV1.Node,
+	capacities []*genV1.CapacityRequest) error {
+	// ACR CRD
+	reservation := genV1.AvailableCapacityReservation{
+		Namespace: namespace,
+		Status:    v1.ReservationRequested,
+	}
+
+	// fill in reservation requests
+	reservation.ReservationRequests = make([]*genV1.ReservationRequest, len(capacities))
+	for i, capacity := range capacities {
+		reservation.ReservationRequests[i] = &genV1.ReservationRequest{CapacityRequest: capacity}
+	}
 
+	// fill in node requests
+	reservation.NodeRequests = &genV1.NodeRequests{}
+	if nodes, err := e.prepareListOfRequestedNodes(nodes); err == nil {
```

Comment:
Can rewrite it in this way
```
if nodes, err := e.prepareListOfRequestedNodes(nodes); err != nil {
	return err
}
reservation.NodeRequests.Requested = nodes
```

Reply:
It will look like this:
```
var (
    nodeIDs []string
    err error
)
if nodeIDs, err = e.prepareListOfRequestedNodes(nodes); err != nil {
    return err
}
reservation.NodeRequests.Requested = nodeIDs
```
Is that ok for you?

Reply:
Yes, that's ok